src/TextField/TextField.js

```diff
@@ -73,6 +76,15 @@ export default class TextField extends Component {
      * The CSS class name of the root element.
      */
     className: PropTypes.string,
+    /**
+     * Whether the label should be displayed in an error state
+     */
+    error: PropTypes.bool,
+    /**
+     * Whether this label should indicate that the input
+     * is required.
+     */
+    required: PropTypes.bool,
```

Comment:
It is not an absolute necessity at the top `TextField`, but it is convenient since `error` must be supplied at this level.  This can be removed but I do think it is nice to have.